yt_dlp/extractor/qqmusic.py

```diff
@@ -346,19 +399,83 @@ def _real_extract(self, url):
             'http://i.y.qq.com/qzone-music/fcg-bin/fcg_ucc_getcdinfo_byids_cp.fcg',
             list_id, 'Download list page',
             query={'type': 1, 'json': 1, 'utf8': 1, 'onlysong': 0, 'disstid': list_id},
-            transform_source=strip_jsonp)
+            transform_source=strip_jsonp, headers={'Referer': url})
         if not len(list_json.get('cdlist', [])):
-            if list_json.get('code'):
-                raise ExtractorError(
-                    'QQ Music said: error %d in fetching playlist info' % list_json['code'],
-                    expected=True)
-            raise ExtractorError('Unable to get playlist info')
-
-        cdlist = list_json['cdlist'][0]
-        entries = [self.url_result(
-            'https://y.qq.com/n/yqq/song/' + song['songmid'] + '.html', 'QQMusic', song['songmid'])
-            for song in cdlist['songlist']]
-
-        list_name = cdlist.get('dissname')
-        list_description = clean_html(unescapeHTML(cdlist.get('desc')))
-        return self.playlist_result(entries, list_id, list_name, list_description)
+            error_msg = ''
+            if list_json.get('code') or list_json.get('subcode'):
+                error_msg = f': Error {list_json.get("code")}-{list_json["subcode"]}: {list_json.get("msg", "")}'
+            raise ExtractorError(f'Unable to get playlist info{error_msg}')
+
+        entries = self._extract_entries(list_json, ('cdlist', 0, 'songlist', ...))
+
+        return self.playlist_result(entries, list_id, **traverse_obj(list_json, ('cdlist', 0, {
+            'title': ('dissname', {str}),
+            'description': ('desc', {lambda x: clean_html(unescapeHTML(x))}),
+        })))
+
+
+class QQMusicVideoIE(QQMusicBaseIE):
+    IE_NAME = 'qqmusic:mv'
+    IE_DESC = 'QQ音乐 - MV'
+    _VALID_URL = r'https?://y\.qq\.com/n/ryqq/mv/(?P<id>[0-9A-Za-z]+)'
+
+    _TESTS = [{
+        'url': 'https://y.qq.com/n/ryqq/mv/002Vsarh3SVU8K',
+        'info_dict': {
+            'id': '002Vsarh3SVU8K',
+            'ext': 'mp4',
+            'title': 'The Chant (Extended Mix / Audio)',
+            'description': '',
+            'thumbnail': r're:^https?://.*\.jpg(?:$|[#?])',
+            'release_timestamp': 1688918400,
+            'release_date': '20230709',
+            'duration': 313,
+            'creators': ['Duke Dumont'],
+            'view_count': int,
+        },
+    }]
+
+    def _parse_url_formats(self, url_data):
+        return traverse_obj(url_data, ('mp4', lambda _, v: v.get('freeflow_url'), {
```

Comment:
don't need to worry about keyerrors when filtering in traversal
```suggestion
        return traverse_obj(url_data, ('mp4', lambda _, v: v['freeflow_url'], {
```